frontend/utils.mjs

```diff
@@ -92,17 +97,25 @@ export function copyIndexHtml(
         window.ESBUILD_LOAD_CHUNKS('index');
     `
 
-    const cssLinkTag = cssFile ? `<link rel="stylesheet" href='${useJsURL ? jsURL : ''}/static/${cssFile}'>` : ''
+    // Snippet to dynamically load the css based on window.JS_URL
+    const cssLoader = `
+        const link = document.createElement("link");
+        link.rel = "stylesheet";
+        link.href = window.JS_URL + "/static/${cssFile}"
+        document.head.appendChild(link)
+    `
 
     fse.writeFileSync(
         path.resolve(__dirname, to),
         fse.readFileSync(path.resolve(__dirname, from), { encoding: 'utf-8' }).replace(
             '</head>',
             `   <script type="application/javascript">
+                    // Load styles first, such that we have them before code
+                    // loads and starts adding elements to html
+                    ${cssLoader}
                     ${scriptCode}
                     ${Object.keys(chunks).length > 0 ? chunkCode : ''}
                 </script>
-                ${cssLinkTag}
```

Comment:
Some context below regarding the order of things.

The CSS link tag (`<link rel="stylesheet" ... />`) that was there before blocks execution of the rest of the page until it finishes loading. When we had this tag before the `<script>` block, the initial page load for the user was sluggish because of a white screen. The page would load the CSS and only then unblock and start loading scripts. Changing the order of things helped dramatically.

It seems like dynamically injecting a CSS link tag does not make it block the same way actually having the tag in the DOM does. Both asset types start loading simultaneously:

![2022-01-27 13 55 54](https://user-images.githubusercontent.com/53387/151363143-dbd748eb-5a2d-424f-a7ec-c3812f9880f6.gif)

However, we now have the danger of running into what's called "the flash of unstyled content". We can end up in a scenario where the CSS is still loading, but all the JS has finished downloading and running... and the app has rendered, just without the CSS. 

I don't know how common this would be, but we should watch out.

Reply:
Yeah I'm not exactly sure on what the order of loading is. I've added a comment regarding this being a risk. Do you think there is anything else to do here?